[quant] set per_channel quant as default for QNNPACK

isdanni opened the pull request:
Summary:
The current default `qconfig` for qnnpack is per-tensor quantization. Since per channel quantization has been supported on qnnpack for a while, setting per channel quant as default qconfig.

**Note**: per channel is for weights only.

Differential Revision: D42005831

Close: #90289

cc: @kimishpatel @vkuzo @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @jgong5 @Xia-Weiwen @leslie-fang-intel

vkuzo reviewed (changes requested):
thanks for doing this!  This is a very nice improvement.

Can we add a version 2 to `get_default_qat_config`, make version 2 the default and keep behavior of versions 0 and 1 as is?  This would enable customers to easily use the old settings if needed.

Can we also add this to `get_default_qconfig`, with a new version?

digantdesai reviewed (approved):
LGTM

isdanni committed:
quantization qconfig: set per_channel quant as default for QNNPACK (#90782)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/90782

The current default `qconfig` for qnnpack is per-tensor quantization. Since per channel quantization has been supported on qnnpack for a while, setting per channel quant as default qconfig.

Following CR from https://github.com/pytorch/pytorch/pull/90782#pullrequestreview-1216341571, adding a new version for `get_default_qconfig` and `get_default_qat_qconfig` respectively and making it the default.

**Note**: per channel is for weights only.

Test Plan: contbuild & OSS CI, see

Reviewed By: digantdesai

Differential Revision: D42005831

fbshipit-source-id: c39c19b6f46fda97a2e8f04c68c0671725051060

vkuzo commented:
do we plan to make the new behavior default?

vkuzo commented:
how about:
1. change the default version to be your new one (this is BC breaking)
2. in the same diff, update Meta callsites to use the previous version (this makes the Meta callsites behave as before)

This will still be BC breaking for non-Meta, but I think it's a good change to make.  However, I'm not sure that we should add a new version without setting it as the default, as that wouldn't actually enable the functionality by default for anyone.

wdyt?

isdanni commented:
> how about:
> 
> 1. change the default version to be your new one (this is BC breaking)
> 2. in the same diff, update Meta callsites to use the previous version (this makes the Meta callsites behave as before)
> 
> This will still be BC breaking for non-Meta, but I think it's a good change to make. However, I'm not sure that we should add a new version without setting it as the default, as that wouldn't actually enable the functionality by default for anyone.
> 
> wdyt?

sounds good.  I will make changes accordingly :)
